cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionFilter.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Describes a filter over partition keys.
+ */
+public class PartitionFilter {
+
+  private final Map<String, Condition<? extends Comparable>> conditions;
+
+  // we only allow creating an filter through the builder.
+  private PartitionFilter(Map<String, Condition<? extends Comparable>> conditions) {
+    this.conditions = conditions;
+  }
+
+  /**
+   * This should be used for inspection or debugging only.
+   * To match this filter, use the {@link #match} method.
+   *
+   * @return the individual conditions of this filter.
+   */
+  public Map<String, Condition<? extends Comparable>> getConditions() {
+    return ImmutableMap.copyOf(conditions);
+  }
+
+  /**
+   * @return the condition for a particular field.
+   */
+  public Condition<? extends Comparable> getCondition(String fieldName) {
+    return conditions.get(fieldName);
+  }
+
+  /**
+   * Match this filter against a partition key. The key matches iff it matches all conditions.
+   *
+   * @throws java.lang.IllegalArgumentException if one of the field types in the partition key are incompatible
+   */
+  public boolean match(PartitionKey partitionKey) {
+    for (Map.Entry<String, Condition<? extends Comparable>> condition : getConditions().entrySet())  {
+      Comparable value = partitionKey.getField(condition.getKey());
+      if (value == null || !condition.getValue().match(value)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public String toString() {
+    return conditions.values().toString();
```

Comment:
The values are Conditions and a condition also has the field name. So this would be redundant. Oh, but perhaps I have to add that to Condition.toString()